src/key_material.rs

```diff
@@ -225,4 +236,101 @@ mod test {
             withdrawal_creds_from_pk(key_material_3.withdrawal_pk.as_ref().unwrap())
         );
     }
+
+    #[test]
+    fn test_seed_to_keystore_scrypt() {
+        let seed = seed_from_mnemonic();
+        let keystores = seed_to_key_material(
+            &seed,
+            3,
+            0,
+            Some(VOTING_KEYSTORE_PASSWORD),
+            false,
+            Some("scrypt"),
+        );
+
+        assert_eq!(keystores.len(), 3);
+
+        // Keys asserted here are generated with
+        // ./deposit existing-mnemonic --eth1_withdrawal_address 0x0000000000000000000000000000000000000001 --keystore_password testtest
+
+        // **[Warning] you are setting an Eth1 address as your withdrawal address. Please ensure that you have control over this address.**
+
+        // Please enter your mnemonic separated by spaces (" "): entire habit bottom mention spoil clown finger wheat motion fox axis mechanic country make garment bar blind stadium sugar water scissors canyon often ketchup
+        // Enter the index (key number) you wish to start generating more keys from. For example, if you've generated 4 keys in the past, you'd enter 4 here. [0]: 0
+        // Please choose how many new validators you wish to run: 3
+        // Please choose the (mainnet or testnet) network/chain name ['mainnet', 'prater', 'kintsugi', 'kiln', 'minimal']:  [mainnet]: minimal
+
+        let key_material_1 = keystores.get(0).unwrap();
+        assert_eq!(key_material_1.clone().keystore.unwrap().pubkey(), "8666389c3fe6ff0bca9adba81504f380b9e2c719419760d561836472fafe295cb50696524e19cba084e1d788d66c80d6");
+        let secret_key = &key_material_1.voting_secret;
+        assert_eq!(
+            "3f3e0a69a6a66aeaec606a2ccb47c703afb2e8ae64f70a1650c03343b06e8f0c",
+            hex::encode(secret_key.as_bytes())
+        );
+
+        let key_material_2 = keystores.get(1).unwrap();
+        assert_eq!(key_material_2.clone().keystore.unwrap().pubkey(), "974ec5bce4653f1f440ad07c5b363ad3b1616520e9680ff837f9ff7a8c10e3cc67dd49aa5089f714ed45d7ad56bc758a");
+        let secret_key = &key_material_2.voting_secret;
+        assert_eq!(
+            "5b17290ac4e36c497378351ed61253a239d408466918fe5a8dd79b28ef67f9df",
+            hex::encode(secret_key.as_bytes())
+        );
+
+        let key_material_3 = keystores.get(2).unwrap();
+        assert_eq!(key_material_3.clone().keystore.unwrap().pubkey(), "8bf0a669a51d0cb6ff745e4b0aa7c41e8de8d179ff9267977e76c7188aaa2fb1b8b1bdfefcc545d9efdac0b4bc2239e6");
+        let secret_key = &key_material_3.voting_secret;
+        assert_eq!(
+            "1cb46382c3c046f7817298d1ad3454a7175be6e53059bdd287e8fcbf4c6fb2e8",
+            hex::encode(secret_key.as_bytes())
+        );
```

Comment:
This should also test that generated keystore has proper encryption specified, e.g.

given generated keystore record is like
```
{
  "crypto": {
    "checksum": {
      "function": "sha256",
      "message": "b95e2bf5c5aafe0e9149f08d2e38f1daa5d20b73972433cd1b523e647249544c"
    },
    "cipher": {
      "function": "aes-128-ctr",
      "message": "bbed72f35ea11f66a44ca42c5b593574db1fb1bd763c91e64a8629899fb58ffe",
      "params": {
        "iv": "793b3201f800b017e80c35b5486ab5df"
      }
    },
    "kdf": {
      "function": "scrypt",
      "message": "",
      "params": {
        "dklen": 32,
        "n": 262144,
        "p": 1,
        "r": 8,
        "salt": "14260c0c72c4afcde2b99ccdf2e1b1257d3cb5dfdc19ddbdc70f2b36eee90b9a"
      }
    }
  },
  "description": "",
  "path": "m/12381/3600/9/0/0",
  "pubkey": "b9d798bbff9a7d7599617035c1141ef607f0b4398ad9c3753219c634397b0a6f6f162b280f9322e41e4878d3d1a351df",
  "uuid": "8e32aff9-96f2-4e87-80c7-0db7ab1f8327",
  "version": 4
}
```

the test logic should parse json and assert /kdf/function property is "scrypt"

Reply:
added in [2064c11](https://github.com/ChorusOne/eth-staking-smith/pull/18/commits/2064c11669605549e442830e780a2dff7ff6f50a)